tstest/integration/integration_test.go

```diff
@@ -795,7 +830,33 @@ func (tt *trafficTrap) Err() error {
 	return nil
 }
 
+//go:embed testdata/rootDesc.xml
+var rootDesc []byte
+
+//go:embed testdata/addAnyPortMapping.xml
+var addAnyPortMapping []byte
+
+//go:embed testdata/getExternalIP.xml
+var getExternalIP []byte
+
 func (tt *trafficTrap) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	if r.URL.EscapedPath() == "/rootDesc.xml" {
```

Comment:
worth some comments about what this is? Or better, pull this into a method with a nice name so it's obvious what it is.

```go
     if tt.serveUPnP(w, r) {
            return
     }

// serveUPnP acts like a UPnP gateway device.
// It reports whether the request r was handled. (That is, whether r looked
// like a UPnP query)
func (tt *trafficTrap) serveUPnP(...) (handled bool) {
    
    return false
}